openai/api_requestor.py

```diff
@@ -294,18 +294,29 @@ async def arequest(
         request_id: Optional[str] = None,
         request_timeout: Optional[Union[float, Tuple[float, float]]] = None,
     ) -> Tuple[Union[OpenAIResponse, AsyncGenerator[OpenAIResponse, None]], bool, str]:
-        async with aiohttp_session() as session:
-            result = await self.arequest_raw(
-                method.lower(),
-                url,
-                session,
-                params=params,
-                supplied_headers=headers,
-                files=files,
-                request_id=request_id,
-                request_timeout=request_timeout,
-            )
-            resp, got_stream = await self._interpret_async_response(result, stream)
+        ctx = aiohttp_session()
+        session = await ctx.__aenter__()
+        result = await self.arequest_raw(
+            method.lower(),
+            url,
+            session,
+            params=params,
+            supplied_headers=headers,
+            files=files,
+            request_id=request_id,
+            request_timeout=request_timeout,
+        )
+        resp, got_stream = await self._interpret_async_response(result, stream)
+        if got_stream:
+
+            async def wrap_resp():
+                async for r in resp:
+                    yield r
+                await ctx.__aexit__(None, None, None)
```

Comment:
I guess it's possible for this async generator to never complete (for example if the caller raises an exception before completion the iterations) in which case we'll never close this session, which I think raises an exception on the event loop?

Maybe we should create a session on the `APIRequestor` instance instead 🤔

Reply:
Good catch, I think should be good now in the latest commit